SeaBattle/Ships/BaseShip.cs

```diff
@@ -0,0 +1,159 @@
+        
+        public bool Equals(BaseShip thisShip)
+        {
+            if (ReferenceEquals(thisShip, null))
+                return false;
+            if (ReferenceEquals(this, thisShip))
+                return true;
+            return Decks.Count.Equals(thisShip.Decks.Count) && GetType().Equals(thisShip.GetType());
+        }
+        public override bool Equals(object s1) => Equals(s1 as BaseShip);
+        // To do in the MoveTo method  
+        // 1. + calculating mid-ship coordinates of current position
+        // 2. + calculating mid-ship coordinates of new position
+        // 3. + calcualating distance of the move
+        // 4. + checking if speed is enough to get to new position
+        // 5. + checking if new position is available
+        // 6. + updating current position with new position
+        public void MoveTo(List<Point> NewPosition, Sea Sea1)
+        {
+            float midShipX = 0;
+            float midShipY = 0;
+            foreach (var d in Decks)
+            {
+                midShipX += d.Location.X;
+                midShipY += d.Location.Y;
+            }
+            midShipX /= Decks.Count;
+            midShipY /= Decks.Count;
+
+            float midNewX = 0;
+            float midNewY = 0;
+            foreach (var p in NewPosition)
+            {
+                midNewX += p.X;
+                midNewY += p.Y;
+            }
+            midNewX /= NewPosition.Count;
+            midNewY /= NewPosition.Count;
+
+            float TravelX = (float)Math.Pow(midShipX - midNewX, 2);
+            float TravelY = (float)Math.Pow(midShipY - midNewY, 2);
+            float TravelDistance;
+            TravelDistance = (float)Math.Sqrt(TravelX + TravelY);
+            if (Speed < TravelDistance)
+            {
+                throw new Exception("Ship can't move");
+            }
+            if (Sea1.IsPlaceAvailable(this) && this.ShipCoordsLine(NewPosition) && this.ShipCoordsUnion(NewPosition) && this.ShipCoordsUnique(NewPosition))
+            {
+                for (int i = 0; i < NewPosition.Count; i++)
+                {
+                    Decks[i].Location.X = NewPosition[i].X;
+                    Decks[i].Location.Y = NewPosition[i].Y;
+                }
+            }
+            else
+                throw new Exception("Ship can't move");
+        }
+        public bool HasCoordinatesIntersect(List<Deck> DecksList)
+        {
+            return Decks.Any(e => e.HasCoordinatesIntersect(DecksList));
+        }
+        public bool ShipCoordsLine(List<Point> SameDecks)
+        {
+            bool lineFlagX = true;
+            bool lineFlagY = true;
+
+            for (int i = 0; i < SameDecks.Count - 1; i++)
+            {
+                lineFlagX &= SameDecks[i].X == SameDecks[i + 1].X;
+                lineFlagY &= SameDecks[i].Y == SameDecks[i + 1].Y;
+            }
+            return (lineFlagX ^ lineFlagY);
```

Comment:
Why do you use brackets here? It's not really a big deal though, but I think when it's not necessary better not to use anything.

Reply:
Fixed.